apps/omg_api/lib/state/transaction.ex

```diff
@@ -54,63 +54,95 @@ defmodule OMG.API.State.Transaction do
   @type currency :: Crypto.address_t()
 
   @doc """
-  Creates transaction from utxos where first output belongs to receiver and second belongs to owner of utxos
-  and the amount decreased by receiver's amount and the fee.
+  Creates transaction from utxo positions and outputs. Provides simple, stateless validation on arguments.
 
-  assumptions:
-   ```length(utxos) = 1 | 2```
+  #### Assumptions:
+   * length of inputs between 1 and @max_inputs
+   * length of outputs between 0 and @max_inputs
+   * the same currency for each output
+   * all amounts are non-negative integers
   """
   @spec create_from_utxos(
-          %{address: Crypto.address_t(), utxos: map()},
-          %{address: Crypto.address_t(), amount: pos_integer()},
-          fee :: non_neg_integer()
+          [
+            %{
+              blknum: pos_integer(),
+              txindex: non_neg_integer(),
+              oindex: 0 | 1,
+              currency: Crypto.address_t(),
+              amount: pos_integer()
+            }
+          ],
+          [%{owner: Crypto.address_t(), amount: non_neg_integer()}],
+          non_neg_integer()
         ) :: {:ok, t()} | {:error, atom()}
-  def create_from_utxos(sender_utxos, receiver, fee \\ 0)
-  def create_from_utxos(_utxos, _receiver, fee) when fee < 0, do: {:error, :invalid_fee}
-  def create_from_utxos(%{utxos: utxos}, _, _) when length(utxos) > @max_inputs, do: {:error, :too_many_utxo}
-
-  def create_from_utxos(%{utxos: utxos} = inputs, receiver, fee) do
-    with {:ok, currency} <- validate_currency(utxos) do
-      do_create_from_utxos(inputs, currency, receiver, fee)
+  def create_from_utxos(inputs, outputs, fee)
```

Comment:
I think the ticket you're mentioning was for removal of the `fee` field for the Transaction. Fee's still there but implicit, as a difference between inputs and outputs value. 
I'm making it a TODO - it would be nice if Watcher could warn that the difference would not cover the fee, but not for now... 
Shall you agree?